File: frontend/src/api/businessDashboard.ts
import { apiRequest } from './http'

export interface BusinessDashboardParams { start?: string; end?: string }

export interface BusinessDashboard {
  start: string
  end: string
  customer_total: number
  funnel: Array<{ stage: string; count: number }>
  orders: { total: number; by_status: Record<string, number>; paid_or_completed: number; paid_amount: number }
  tickets: { total: number; by_status: Record<string, number>; open_count: number }
  paying_customer_count: number
  repeat_purchase_customer_count: number
  renewal_rate: number
  renewal_rate_definition: string
  owner_efficiency: Array<{
    user_id: number
    username: string
    full_name: string | null
    customer_count: number
    converted_customer_count: number
    paid_amount: number
    open_ticket_count: number
  }>
}

export function getBusinessDashboard(params: BusinessDashboardParams = {}): Promise<BusinessDashboard> {
  const query = new URLSearchParams()
  if (params.start) query.set('start', params.start)
  if (params.end) query.set('end', params.end)
  const suffix = query.toString() ? `?${query.toString()}` : ''
  return apiRequest<BusinessDashboard>(`/admin/business-dashboard${suffix}`)
}

File: backend/app/schemas/business_dashboard.py
from datetime import datetime

from pydantic import BaseModel


class BusinessFunnelStage(BaseModel):
    stage: str
    count: int


class BusinessOrderMetrics(BaseModel):
    total: int
    by_status: dict[str, int]
    paid_or_completed: int
    paid_amount: float


class BusinessTicketMetrics(BaseModel):
    total: int
    by_status: dict[str, int]
    open_count: int


class BusinessOwnerMetric(BaseModel):
    user_id: int
    username: str
    full_name: str | None
    customer_count: int
    converted_customer_count: int
    paid_amount: float
    open_ticket_count: int


class BusinessDashboardRead(BaseModel):
    start: datetime
    end: datetime
    customer_total: int
    funnel: list[BusinessFunnelStage]
    orders: BusinessOrderMetrics
    tickets: BusinessTicketMetrics
    paying_customer_count: int
    repeat_purchase_customer_count: int
    renewal_rate: float
    renewal_rate_definition: str
    owner_efficiency: list[BusinessOwnerMetric]
